src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -232,7 +232,7 @@ public function getResources()
      */
     public function addResource(ResourceInterface $resource)
     {
-        $this->resources[$resource->__toString()] = $resource;
+        $this->resources[(string) $resource] = $resource;
```

Comment:
Point is not to use magic methods directly (as magic methods are PHP internals).